### ISSUE TITLE
Compiler IDs for aliases not working in client coe

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer#772]
Looks like an issue where if a URL has a pathname instead of an ID then the client code can't find it in the list of compilers. We'll have to support aliases too.

https://sentry.io/compiler-explorer/compiler-explorer/issues/434963846/

```
Unable to find compiler id "/opt/clang+llvm-3.4.1-x86_64-unknown-ubuntu12.04/bin/clang++"
```

**Comment by mattgodbolt** [comment]
Duplicate of #771

**Comment by RabsRincon** [comment]
b5f0596b should have fixed this, but will keep an eye out for a bit

**Comment by RabsRincon** [comment]
Not fixed, we're still getting 404s

**Comment by mattgodbolt** [comment]
This issue is not necessarily related to the 404s. The 404s are server-side, this bug was an error happening on the client.

**Comment by RabsRincon** [comment]
Fixed by 4e5aef0db1fe58c7383b99b349cbe858f8b819e2
